src/envs/pddl_env.py

```diff
@@ -0,0 +1,470 @@
+        return Box(lb, ub, dtype=np.float32)
+
+    def render_state(self,
+                     state: State,
+                     task: Task,
+                     action: Optional[Action] = None,
+                     caption: Optional[str] = None) -> List[Image]:
+        raise NotImplementedError("Render not implemented for PDDLEnv.")
+
+
+class _FixedTasksPDDLEnv(_PDDLEnv):
+    """An environment where tasks are induced by static PDDL problem files."""
+
+    @property
+    @abc.abstractmethod
+    def _pddl_problem_asset_dir(self) -> str:
+        """The name of the problem file directory in assets/pddl."""
+        raise NotImplementedError("Override me!")
+
+    @property
+    @abc.abstractmethod
+    def _train_problem_indices(self) -> List[int]:
+        """Return a list of indices corresponding to training problems.
+
+        For each idx in the returned list, it is expected that there
+        will be a problem file named task{idx}.pddl.
+        """
+        raise NotImplementedError("Override me!")
+
+    @property
+    @abc.abstractmethod
+    def _test_problem_indices(self) -> List[int]:
+        """Return a list of indices corresponding to test problems.
+
+        For each idx in the returned list, it is expected that there
+        will be a problem file named task{idx}.pddl.
+        """
+        raise NotImplementedError("Override me!")
+
+    @property
+    def _pddl_train_problem_generator(self) -> PDDLProblemGenerator:
+        assert len(self._train_problem_indices) >= CFG.num_train_tasks
+        return _file_problem_generator(self._pddl_problem_asset_dir,
+                                       self._train_problem_indices)
+
+    @property
+    def _pddl_test_problem_generator(self) -> PDDLProblemGenerator:
+        assert len(self._test_problem_indices) >= CFG.num_test_tasks
+        return _file_problem_generator(self._pddl_problem_asset_dir,
+                                       self._test_problem_indices)
+
+
+###############################################################################
+#                              Specific Envs                                  #
+###############################################################################
+
+
+class _BlocksPDDLEnv(_PDDLEnv):
+    """The IPC 4-operator blocks world domain."""
+
+    @property
+    def _domain_str(self) -> str:
+        path = utils.get_env_asset_path("pddl/blocks/domain.pddl")
+        with open(path, encoding="utf-8") as f:
+            domain_str = f.read()
+        return domain_str
+
+
+class FixedTasksBlocksPDDLEnv(_BlocksPDDLEnv, _FixedTasksPDDLEnv):
+    """The IPC 4-operator blocks world domain with a fixed set of tasks."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "pddl_blocks_fixed_tasks"
+
+    @property
+    def _pddl_problem_asset_dir(self) -> str:
+        return "blocks"
+
+    @property
+    def _train_problem_indices(self) -> List[int]:
+        return list(range(1, 6))
```

Comment:
fine with me, but let's at least move this into settings.py? `pddl_blocks_fixed_train_indices` and `pddl_blocks_fixed_test_indices` for instance